### PR TITLE
Implement API routes

### PR DESCRIPTION
**Pull request by tf3**
Implement https://trello.com/c/Ewe4On1t/11-api-routes-l

CircleCI tests are failing, but that's just because I haven't yet gotten it configured to work with MongoDB. On my local machine, the tests are passing.

### CODE REVIEW

**Comment by mmcphillips** [comment]
Looks good from what I've looked over. I have a question about 
067e5b8. What would be the purpose of getting mongoose to utilize bluebird promises? It was my understanding that mongoose queries generally returned promises?

**Comment by tf3** [comment]
> Looks good from what I've looked over. I have a question about
> [067e5b8](https://github.com/fec4-gandolf/bid-buy/commit/067e5b87da6381f3592ed839070a1f20aa30ccdb). What would be the purpose of getting mongoose to utilize bluebird promises? It was my understanding that mongoose queries generally returned promises?

Mongoose queries do return "thenables". But I'm using Bluebird because its promises have features that the Mongoose thenables lack by default (for example, a `tap()` method).

File: server/index.test.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+const rpOptionsPost = (id, bid) => {
+  return {
+    method: 'POST',
+    url: `http://localhost:3001/bid/${id}`,
+    body: {bid},
+    json: true
+  };
+};
+
+// The database should be seeded with `npm run seed` before these tests are run.
```
Comment (NapoleonDLP) [inline server/index.test.js]: Would these instructions useful in the README?

Comment (NapoleonDLP) [reply server/index.test.js]: Actually, now that I think about it. This is for testing. Testing happens in development. No need to add to README

Comment (tf3) [reply server/index.test.js]: Hmm, thanks. It might be worth making a note about it at some point.

**Pull request merged**